lib/metasploit/framework/data_service/proxy/loot_data_proxy.rb

```diff
@@ -4,7 +4,10 @@ def report_loot(opts)
     begin
       self.data_service_operation do |data_service|
         if !data_service.is_a?(Msf::DBManager)
-          opts[:data] = Base64.urlsafe_encode64(opts[:data].empty? ? "" : opts[:data].join('')) if opts[:data] and opts[:data].kind_of?(Array) else opts[:data]
+          unless opts[:data].nil?
+            opts[:data] = opts[:data].join if opts[:data].kind_of?(Array)
+            opts[:data] = Base64.urlsafe_encode64(opts[:data]) unless opts[:data].empty?
```

Comment:
So my one concern is that in the previous code if `opts[:data]` is empty, then we will still call `Base64.urlsafe_encode64` on it. Your current code doesn't do this, but I'm not sure if its needed or not.

Reply:
For the empty string there is no difference
```
irb(main):003:0> Base64.encode64 ""
=> ""
irb(main):004:0> Base64.decode64 ""
=> ""
```